cli/azd/pkg/templates/source_manager.go

```diff
@@ -234,7 +231,16 @@ func (sm *sourceManager) CreateSource(ctx context.Context, config *SourceConfig,
 	case SourceKindResource:
 		source, err = newJsonTemplateSource(SourceDefault.Name, string(resources.TemplatesJson))
 	case SourceKindGh:
-		source, err = newGhTemplateSource(ctx, config.Name, config.Location, sm.ghCli, console)
+		// resolve the GitHub cli from container
+		var dependencies struct {
+			ghCli   *github.Cli   `container:"type"`
+			console input.Console `container:"type"`
+		}
+		err = sm.serviceLocator.Fill(&dependencies)
```

Comment:
I feel like decorating the types on `gh_source.go` and just calling `sm.serviceLocator.Fill(&ghSource)` would be slightly better -- then the source manager would "truly not know about dependencies being exposed from a source implementation".

If we are still thinking along the lines proposed here, one alternative here is:
```golang
err = container.Call(func (ghCli *github.Cli, console input.Console) error {
   source, err = newGhTemplateSource(ctx, config.Name, config.Location, ghCli, console)
   return err
})
```

Reply:
updated it to use serviceLocator.Invoke()